src/pages/index.js

```diff
@@ -67,250 +74,229 @@ const cardList = new Section({
- // imagePopup.close()
-  //closePopUp(pictureModal)
-//}) //Close icons should be handled only once in the file body, otherwise you add listeners again and again to the same elements. This can cause a memory leak
 
+  addFormValidator.disabledButton();
 
-//close modal if clicked outside forms
-/*
-modals.forEach((modal) => {
-  modal.addEventListener('click', (e) => {
-    if (e.target == modal) {
-      closePopUp(e.target)
-      // closePopUp(profileEditModal) or any other modal
-    }
-  })
-})
-  */
-//if you click escape close the modals
-/*
-function escapeModalHandler(e) {
-  if (e.key === "Escape") {
-    const openedPopup = document.querySelector('.modal_opened')
-    closePopUp(openedPopup)
-  }
 }
-*/
-
-/*
-document.querySelector('body').addEventListener('keydown', (e) => {
-  if (e.key === "Escape") {
-    closePopUp(profileEditModal)
-    closePopUp(pictureModal)
-    closePopUp(cardAddModal)
 
-  }
-})
-*/
-
-// TODO define handle image click function
 function handleImageClick(name, link) {
-  //pictureModalImage.setAttribute('src', link) //go ot the image element and change it to the image source that is  in line 69
-  //pictureModalImage.alt = name; //set the alt text to the name
-  //pictureModalCaption.textContent = /*cardData.*/ name // switch the alternate text of the modal
-  imagePopup.open({name,link})
- // openPopUp(pictureModal)
-
+  imagePopup.open({
+    name,
+    link,
+  });
 }
 
-//places each card into the list in the DOM
-
-//initialCards.forEach((cardData) => {
-  // const cardElement = renderCard(cardData); //get each obj and store it into a variable that will be rendered
-  //cardGallerySection.append(cardElement);//apends to the end of the gallery also was used in rendering function
-  //const cardView = getCardView(cardData) //get the cardview for each particular object element and store it into a variable
-  //i need to use the card element from the card class that and the selector of the template
-  // TODO pass a third argument, a function to handle image click
-  //const card = new Card(cardData, '.card-template', handleImageClick)
- // const cardView = /*card.*/ getCardView(cardData);
-  //TODO call cardview function for avoid repeating yourself.
- // renderCard(cardView, cardGallerySection) //needs to pass card which is the object iterations and the element in which it will be appended in the gallery
-//});
-
-/*
-const cardLikeBtn = document.querySelectorAll(".card__heart")
-cardLikeBtn.forEach((heart)=>{
-  heart.addEventListener("click", ()=>{
-    heart.classList.toggle('card__heart_active')
-  })
-})
-*/
+//needs access to the to the id of the card and the card element
+//we want to make use of a closure
+function handleDeleteButton(id, cardElement) {
+  areYouDeletingCardPopup.open();
+
+  function handleDeleteSubmit() {
+    // need access to card ID and card element
+    api.deleteACard(id).then(() => {
+      cardElement.remove();
+      areYouDeletingCardPopup.close();
+    })
```

Comment:
You need to `catch` possible errors at the very **end** of any server request.

**Please, correct it everywhere in the code**